frontend/tests/controllers/QualityNominationTest.php

```diff
@@ -2451,4 +2451,144 @@ public function testCanonicalizeTags() {
             ['lenguaje', 'problemTopicDynamicProgramming', 'problemTopicGeometry', 'problemTopicMath', 'problemTopicSorting']
         );
     }
+
+    /**
+     * A PHPUnit data provider for all the tests that can accept a status.
+     *
+     * @return list<array{0: string, 1: int, 2: int, 3: string, 4:array<int:boolean>, 5: boolean }>
+     */
+    public function qualityNominationsDemotionStatusApiUpdateCaseProvider(): array {
+        return [
+            ['warning', \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING,
+            'qualityNominationProblemHasWarning',
+            [\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_BANNED => false,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED => false,
+            \OmegaUp\ProblemParams::VISIBILITY_PRIVATE => false,\OmegaUp\ProblemParams::VISIBILITY_PRIVATE_WARNING => true,
+            \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING => true,\OmegaUp\ProblemParams::VISIBILITY_PUBLIC => false,
+            \OmegaUp\ProblemParams::VISIBILITY_PROMOTED => false], true
+            ],
```

Comment:
ándale, me gustó cómo lo organizaste. sólo puedes darle un poco de formato para que se entienda mejor?

```suggestion
            [
                'warning',
                \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING,
                'qualityNominationProblemHasWarning',
                [
                     \OmegaUp\ProblemParams::VISIBILITY_PRIVATE_BANNED => false,
                     \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_BANNED => false,
                     \OmegaUp\ProblemParams::VISIBILITY_PRIVATE => false,
                     \OmegaUp\ProblemParams::VISIBILITY_PRIVATE_WARNING => true,
                     \OmegaUp\ProblemParams::VISIBILITY_PUBLIC_WARNING => true,
                     \OmegaUp\ProblemParams::VISIBILITY_PUBLIC => false,
                     \OmegaUp\ProblemParams::VISIBILITY_PROMOTED => false,
                 ],
                 true
            ],
```

Reply:
listo